test/TesterInternal/StreamingTests/DynamicStreamProviderConfigurationTests.cs

```diff
@@ -0,0 +1,327 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+using Orleans;
+using Orleans.Providers;
+using Orleans.Runtime.Configuration;
+using Orleans.Runtime;
+using Orleans.Streams;
+using Orleans.TestingHost;
+using Tester.TestStreamProviders;
+using Orleans.Providers.Streams.Generator;
+using Orleans.TestingHost.Utils;
+using TestGrainInterfaces;
+using TestGrains;
+using Tester;
+using UnitTests.Grains;
+using UnitTests.Tester;
+
+namespace UnitTests.StreamingTests
+{
+    public class DynamicStreamProviderConfigurationTests : OrleansTestingBase, IClassFixture<DynamicStreamProviderConfigurationTests.Fixture>, IDisposable
+    {
+        private Fixture fixture;
+        private static readonly TimeSpan Timeout = TimeSpan.FromSeconds(30);
+        private IManagementGrain mgmtGrain;
+        private const string streamProviderName1 = "GeneratedStreamProvider1";
+        private const string streamProviderName2 = "GeneratedStreamProvider2";
+
+        public class Fixture : BaseTestClusterFixture
+        {
+            public const string StreamNamespace = GeneratedEventCollectorGrain.StreamNamespace;
+            public Dictionary<string, string> DefaultStreamProviderSettings = new Dictionary<string, string>();
+
+            public readonly static SimpleGeneratorConfig GeneratorConfig = new SimpleGeneratorConfig
+            {
+                StreamNamespace = StreamNamespace,
+                EventsInStream = 100
+            };
+
+            public readonly static GeneratorAdapterConfig AdapterConfig = new GeneratorAdapterConfig(GeneratedStreamTestConstants.StreamProviderName)
+            {
+                TotalQueueCount = 4,
+                GeneratorConfigType = GeneratorConfig.GetType()
+            };
+
+            protected override TestCluster CreateTestCluster()
+            {
+                var options = new TestClusterOptions(1);
+                // get initial settings from configs
+                AdapterConfig.WriteProperties(DefaultStreamProviderSettings);
+                GeneratorConfig.WriteProperties(DefaultStreamProviderSettings);
+
+                // add queue balancer setting
+                DefaultStreamProviderSettings.Add(PersistentStreamProviderConfig.QUEUE_BALANCER_TYPE, StreamQueueBalancerType.DynamicClusterConfigDeploymentBalancer.ToString());
+
+                // add pub/sub settting
+                DefaultStreamProviderSettings.Add(PersistentStreamProviderConfig.STREAM_PUBSUB_TYPE, StreamPubSubType.ImplicitOnly.ToString());
+
+                return new TestCluster(options);
+            }
+        }
+
+        public async void Dispose()
+        {
+            await RemoveAllProviders();
+            fixture.DefaultStreamProviderSettings.Remove(FailureInjectionStreamProvider.FailureInjectionModeString);
```

Comment:
This is odd. Please do no update the fixture itself, as it is mutable state across tests and would break as soon as we add some test parallelization (#1863)